ledger/nomos-ledger/src/mantle/sdp/mod.rs

```diff
@@ -302,6 +313,22 @@ impl SdpLedger {
         Ok(self)
     }
 
+    pub fn apply_genesis_declare_msg(
+        mut self,
+        op: &SDPDeclareOp,
+        _config: &Config,
+    ) -> Result<Self, Error> {
+        let declaration_id = op.id();
+        let declaration = Declaration::new(self.block_number, op);
+        if let Some(service_state) = self.services.get_mut(&op.service_type) {
+            service_state.declare_genesis(declaration_id, declaration)?;
+        } else {
+            return Err(Error::ServiceNotFound(op.service_type));
+        }
+
+        Ok(self)
+    }
```

Comment:
I'd prefer a `from_genesis_tx` method (and remove the empty `new`) like in other parts of the ledger so that it's better typed